src/datadog/remote_config/remote_config.cpp

```diff
@@ -173,28 +159,41 @@ std::vector<ConfigMetadata> RemoteConfigurationManager::process_response(
     if (client_configs_it == json.cend()) {
       if (!applied_config_.empty()) {
         std::for_each(applied_config_.cbegin(), applied_config_.cend(),
-                      [this, &config_updates](const auto it) {
-                        auto updated = revert_config(it.second);
-                        config_updates.insert(config_updates.end(),
-                                              updated.begin(), updated.end());
+                      [this](const auto& it) {
+                        for (const auto& listener :
+                             listeners_per_product_[it.second.product]) {
+                          listener->on_revert(it.second);
```

Comment:
`on_revert` is not marked `noexception`, yet here its exceptions can bubble freely. not that rcte2 doesn't provide a way to report exceptions when an error occurs while reverting, so arguably they should be reported globally.

Reply:
`noexception` specifier is not a compile-time check guaranteeing a function is not throwing. It is the responsibility of the implementer to ensure no exceptions are thrown, otherwise, it will result in a call to `std::terminate`, which we cannot recover.  Here is an example:

```cpp
int do_stuff() {
   std::vector<int> v;
   return  v.at(1);
}

void f() noexcept {
   auto a = do_stuff();
}
```

All calls to `on_revert` are in a try-catch block, any exception from the user code will be catched. However I do agree I need to catch `std::exception` for that.

Reply:
I should have been clearer; what I meant was that I see a virtual function not marked `noexcept`, and unless told otherwise, I assume that its implementations can throw (if they can't, why wouldn't they be marked `noexcept`, thereby generating better machine code). Yet the calling code is not prepared for exceptions from the implementations.

Reply:
This has been addressed in edd5397dc1140322035c7f9af2f4f45e3584cf2c